src/controllers/adminStatistics.ts

```diff
@@ -0,0 +1,64 @@
+import { Request, Response } from "express";
+import { sendResponse } from "../utils/http.exception";
+import jwt, { JwtPayload } from "jsonwebtoken";
+import { User } from "../database/models/User";
+import { Role } from "../database/models/role";
+import { Sales } from "../database/models/sales";
+import { Product } from "../database/models/product";
+
+export const adminStatistics = async (req: Request, res: Response) => {
+	try {
+		const admin_token = req.headers.authorization?.split(" ")[1];
+		if (admin_token) {
+			const decoded_admin_token = jwt.decode(admin_token) as JwtPayload;
+			const { id } = decoded_admin_token;
+			if (id) {
+				const SellerID = await Role.findOne({ where: { roleName: "SELLER" } });
+				const buyerID = await Role.findOne({ where: { roleName: "BUYER" } });
+				const allSellers = await User.findAndCountAll({
+					where: { role: SellerID?.dataValues.id },
+				});
+				const allBuyers = await User.findAndCountAll({
+					where: { role: buyerID?.dataValues.id },
+				});
+				const activeBuyers = allBuyers.rows.filter((item) => item.isActive);
+				const activeSellers = allSellers.rows.filter((item) => item.isActive);
+				const allOrders = await Sales.findAndCountAll({include: [

```

Comment:
Instead of using Sales, use Orders: because we can have one order which is corresponding to many sales